src/common/components/signUpModal/signUpModal.component.js

```diff
@@ -636,6 +649,52 @@ class SignUpModalController {
         this.loadingDonorDetails = false
       })
   }
+
+  saveDonorDetails () {
+    const signUpDonorDetails = {
+      name: {
+        'given-name': this.$scope.firstName,
+        'family-name': this.$scope.lastName
+      },
+      'donor-type': this.$scope.accountType,
+      'organization-name': this.$scope.organizationName,
+      email: this.$scope.email,
+      phone: this.$scope.primaryPhone,
+      mailingAddress: {
+        streetAddress: this.$scope.streetAddress,
+        streetAddressExtended:  this.$scope.streetAddressExtended,
+        internationalAddressLine3:  this.$scope.internationalAddressLine3,
+        internationalAddressLine4:  this.$scope.internationalAddressLine4,
+        locality: this.$scope.city,
+        region: this.$scope.state,
+        postalCode: this.$scope.zipCode,
+        country: this.$scope.countryCode
+      }
+    }
+
+    if (angular.isDefined(this.getDonorDetailsSubscription)) {
+      this.getDonorDetailsSubscription.unsubscribe()
+    }
+    this.getDonorDetailsSubscription = this.orderService.getDonorDetails().switchMap((donorDetails) => {
```

Comment:
Did you want to remove this logic from `src/common/components/registerAccountModal/registerAccountModal.component.js` or are you doing that later?

Reply:
I will clean this up in a later PR